backend/app/training/scrape_products.py

```
@@ -0,0 +1,259 @@
+        return False
+    sample_size = min(sample_size, len(urls))
+    sample_urls = random.sample(urls, sample_size)
+    product_count = 0
+    for url in sample_urls:
+        if await is_product_page(url, session):
+            product_count += 1
+    product_ratio = product_count / sample_size
+    print(f"Sampled {sample_size} URLs, found {product_count} product pages (ratio: {product_ratio:.2f})\n")
+    return product_ratio > 0
+
+async def extract_sitemap_urls(sitemap_url, session, check_products=True, sample_size=5):
```

Ruff (0.8.2)

`extract_sitemap_urls` is too complex (11 > 10)

(C901)